### PR TITLE
Implement `documents.update`

### LINKED ISSUE #38: Introduce `documents.update` API
See here for reference: https://www.getoutline.com/developers#tag/Documents/paths/~1documents.update/post

I'd say we implement it the same way as done with documents.create

### PR DESCRIPTION
**Pull request by zadjadr**
Closes #38

### CODE REVIEW

File: package_test.go [hunk 1]
```diff
@@ -364,3 +398,28 @@ const exampleDocumentsCreateResponse_1documents string = `{
 		"deletedAt": "2019-08-24T14:15:22Z"
 	}
 }`
+
+const exampleDocumentsUpdateResponse_1documents string = `{
```
Comment (StefMa) [inline package_test.go:402]: This is a normal `document` right?
Why not re-using the existing fake `document`?

Comment (zadjadr) [reply package_test.go]: True, thanks :+1:

File: documents.go [hunk 2]
```diff
@@ -133,3 +133,75 @@ func (cl *DocumentsCreateClient) Do(ctx context.Context) (*Document, error) {
...
+type DocumentsUpdateClient struct {
+	sl     *sling.Sling
+	params documentsUpdateParams
+}
+
+func newDocumentsUpdateClient(sl *sling.Sling, params documentsUpdateParams) *DocumentsUpdateClient {
+	copy := sl.New()
+
+	return &DocumentsUpdateClient{sl: copy, params: params}
+}
```
Comment (StefMa) [inline documents.go:157]: [Here](https://github.com/ioki-mobility/go-outline/pull/29#discussion_r1282424980) I got told from @rsjethani that we should create the `Params` in this function and put the required (API) parameters as (function) parameters. In this case, the `DocumentId`.

I guess we should do the same here.

File: documents.go [hunk 3]
```diff
@@ -133,3 +133,75 @@ func (cl *DocumentsCreateClient) Do(ctx context.Context) (*Document, error) {
...
+	cl.params.Publish = publish
+	return cl
+}
+
+func (cl *DocumentsUpdateClient) Append(append bool) *DocumentsUpdateClient {
+	cl.params.Append = append
+	return cl
+}
+
+func (cl *DocumentsUpdateClient) Done(done bool) *DocumentsUpdateClient {
```
Comment (StefMa) [inline documents.go:185]: This parameter is marked as
> This property will soon be deprecated.

Not sure if we still want to support it or not 🤔

File: package_test.go [hunk 4]
```diff
@@ -235,9 +234,44 @@ func TestDocumentsClientCreate(t *testing.T) {
...
+	hc := &http.Client{}
+	hc.Transport = &testutils.MockRoundTripper{RoundTripFn: func(r *http.Request) (*http.Response, error) {
+		// Assert request method and URL.
+		assert.Equal(t, http.MethodPost, r.Method)
+		u, err := url.JoinPath(testBaseURL, common.DocumentsUpdateEndpoint())
+		require.NoError(t, err)
+		assert.Equal(t, u, r.URL.String())
+
+		testAssertHeaders(t, r.Header)
+		testAssertBody(t, r, fmt.Sprintf(`{"id":"%s", "title":"%s", "text":"%s", "publish":%t}`, "497f6eca-6276-4993-bfeb-53cbbbba6f08", "🎉 Welcome to Acme Inc", "Updated text!", true))
```
Comment (StefMa) [inline package_test.go:252]: I guess this line is a *bit* too long 😁 
Can we splt it up please 🙏

File: package_test.go [hunk 5]
```diff
@@ -235,9 +234,44 @@ func TestDocumentsClientCreate(t *testing.T) {
...
+			Request:       r,
+			ContentLength: -1,
+			StatusCode:    http.StatusOK,
+			Body:          io.NopCloser(strings.NewReader(testResponse)),
+		}, nil
+	}}
+
+	cl := outline.New(testBaseURL, hc, testApiKey)
+	var id outline.DocumentID = "497f6eca-6276-4993-bfeb-53cbbbba6f08"
+	got, err := cl.Documents().Update(id).Title("🎉 Welcome to Acme Inc").Text("Updated text!").Publish(true).Do(context.Background())
```
Comment (StefMa) [inline package_test.go:264]: Why isn't this inlined? 🤔 
```suggestion
	got, err := cl.Documents().Update(outline.DocumentID("497f6eca-6276-4993-bfeb-53cbbbba6f08")).Title("🎉 Welcome to Acme Inc").Text("Updated text!").Publish(true).Do(context.Background())
```

File: package_test.go [hunk 6]
```diff
@@ -340,7 +374,7 @@ const exampleCollectionsListResponse_1collection string = `
   }
 }`
 
-const exampleDocumentsCreateResponse_1documents string = `{
+const exampleDocumentsResponse_1documents string = `{
```
Comment (StefMa) [inline package_test.go:377]: I guess the naming (`_1documents`) doesn't make sense.
An "document response" contains always just one document.

What about just using `exampleDocumentResponse`?

**Review by StefMa — CHANGES_REQUESTED** [review]
Overall the code loks good 👍

File: package_test.go (hunk 4 above)
Comment (zadjadr) [reply package_test.go:252]: I'll give it a try and check if gofmt changes it back or not

File: documents.go (hunk 3 above)
Comment (zadjadr) [reply documents.go:185]: Yeah, was also not sure if we want to support that - but I think we should support what they support and remove what they remove (?) If that makes sense. I'd just go with whatever they currently provide and remove it when they do

File: documents.go (hunk 2 above)
Comment (zadjadr) [reply documents.go:157]: Very good, thanks - I'll fix that :+1:

Comment (zadjadr) [reply documents.go]: I've also changed that for documents create in a separate commit, but we can also separate it into it's own PR

File: documents.go (hunk 3 above)
Comment (StefMa) [reply documents.go:185]: Yep, fine by me 🙃

**Review by StefMa — APPROVED** [review]
LGTM 👌
Thanks!

**Comment by rsj-ioki** [comment]
@zadjadr please rebase ;)

File: documents.go [hunk 7]
```diff
@@ -78,16 +78,16 @@ type DocumentsCreateClient struct {
 	params documentsCreateParams
 }
 
-func newDocumentsCreateClient(sl *sling.Sling, params documentsCreateParams) *DocumentsCreateClient {
+func newDocumentsCreateClient(sl *sling.Sling, title string, collectionId CollectionID) *DocumentsCreateClient {
```
Comment (rsjethani) [inline documents.go:81]: collectionId -> id; the type shows the param is representing

File: documents.go [hunk 8]
```diff
@@ -133,3 +133,75 @@ func (cl *DocumentsCreateClient) Do(ctx context.Context) (*Document, error) {
...
+	copy := sl.New()
+	params := documentsUpdateParams{Id: id}
+	return &DocumentsUpdateClient{sl: copy, params: params}
+}
+
+// Update returns a client for updating a single document in the specified collection.
+// API reference: https://www.getoutline.com/developers#tag/Documents/paths/~1documents.update/post
+func (cl *DocumentsClient) Update(id DocumentID) *DocumentsUpdateClient {
+	return newDocumentsUpdateClient(cl.sl, id)
+}
```
Comment (rsjethani) [inline documents.go:163]: Please move this above, with other `DocumentsClient` methods

File: package_test.go [hunk 9]
```diff
@@ -427,7 +486,7 @@ const exampleCollectionsListResponse_1collection string = `
   }
 }`
 
-const exampleDocumentsCreateResponse_1documents string = `{
+const exampleDocumentsResponse string = `{
```
Comment (rsjethani) [inline package_test.go:489]: Documents -> Document

**Review by rsjethani — CHANGES_REQUESTED** [review]
Please add the `_failed` test also, otherwise just some cosmetic changes

File: package_test.go (hunk 9 above)
Comment (StefMa) [reply package_test.go]: Fixed here https://github.com/ioki-mobility/go-outline/pull/40/commits/527e85809ecde4a0b2819998d2f05b7383ff1545

File: documents.go (hunk 7 above)
Comment (StefMa) [reply documents.go:93]: Fixed here https://github.com/ioki-mobility/go-outline/pull/40/commits/d87d1c4e9af40811acd4c67a2fe0ee9ad2b0aad4

**Comment by StefMa** [comment]
> Please add the `_failed` test

Fixed here https://github.com/ioki-mobility/go-outline/pull/40/commits/befbbf039b34032d56b6e3686543bcdb22e6cac7

**Review by StefMa — APPROVED** [review]
I should have approved this PR, not [only commenting](https://github.com/ioki-mobility/go-outline/pull/40#pullrequestreview-1567165266) 😁 

However, LGTM 👌 

@zadjadr @rsjethani 
I did the requested changes to finnaly get this thing merged. 
Please have a look 🚀 
(The whole thing will be of course rebased in case you approve 😉 )

**Comment by zadjadr** [comment]
> I should have approved this PR, not [only commenting](https://github.com/ioki-mobility/go-outline/pull/40#pullrequestreview-1567165266) 😁
> 
> However, LGTM 👌
> 
> @zadjadr @rsjethani I did the requested changes to finnaly get this thing merged. Please have a look 🚀 (The whole thing will be of course rebased in case you approve 😉 )

Thanks @StefMa, sorry had a pretty packed week! LGTM :+1:

**Review by rsjethani — APPROVED** [review]
Thank you guys!

**Pull request merged**